### PR TITLE
UserAdmin: SuperUserTrainingList: add support for searching trainings by training creator name

### PR DESCRIPTION
**Pull request by maxime-poulain**
Now super users can filter when searching by a trainer's name.

https://smartjira.atlassian.net/browse/CFA-426

### CODE REVIEW

File: src/UserAdmin/src/Smart.FA.Catalog.Application/UseCases/Queries/GetTrainingsByCriteriaQuery.cs [hunk 1]
```diff
@@ -61,8 +63,26 @@ public async Task<PagedList<Training>> Handle(GetTrainingsByCriteriaQuery reques
...
+            query = query.Where(training => training.TrainerAssignments.Any(assignment => assignment.Trainer.Name.FirstName.Contains(request.TrainerName) || assignment.Trainer.Name.LastName.Contains(request.TrainerName)));
+        }
+
+        // Finally apply pagination.
         var trainings = await query.PaginateAsync(new PageItem(request.PageNumber, request.PageSize), cancellationToken);
 
         return trainings;
     }
+
+    private IQueryable<Training> BaseQuery()
```
Comment (LievenPraet) [inline src/UserAdmin/src/Smart.FA.Catalog.Application/UseCases/Queries/GetTrainingsByCriteriaQuery.cs:78]:
```suggestion
    private IQueryable<Training> GetBaseQuery()
```

**Review by LievenPraet — CHANGES_REQUESTED** [review]

File: src/UserAdmin/src/Smart.FA.Catalog.Application/UseCases/Queries/GetTrainingsByCriteriaQuery.cs (hunk 1 above)
Comment (maxime-poulain) [reply src/UserAdmin/src/Smart.FA.Catalog.Application/UseCases/Queries/GetTrainingsByCriteriaQuery.cs:78]: I tend to put a "Get" prefix on method that gonna fetch one way or another some kind of resource.
"GetTrainerById" -> from db
"GetCountryList" => from an API or so.
To me, it makes it clear that we are fetching something and there's an underlying IO operation under the hood.

I might turn that into a public get property however ?

Comment (LievenPraet) [reply src/UserAdmin/src/Smart.FA.Catalog.Application/UseCases/Queries/GetTrainingsByCriteriaQuery.cs:78]: To me, a function should represent an action: get, insert, delete, convert, send...
IN this case, a property might be a better choice here

**Review by LievenPraet — CHANGES_REQUESTED** [review]
See comments above

**Review by Victor-van-Duijnen — APPROVED** [review]

**Pull request merged**